src/PDPVerifier.sol

```diff
@@ -130,6 +130,8 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
     mapping(uint256 => uint256) challengeRange;
     // Enqueued piece ids for removal when starting the next proving period
     mapping(uint256 => uint256[]) scheduledRemovals;
+    // Track which pieces are scheduled for removal with a bitmap
+    mapping(uint256 dataSetId => uint256 scheduledRemovalsBitmap) scheduledRemovalsBitmap;
```

Comment:
it should still be a two level map because there can be more than 256 pieces. Shift the pieceId down 8 bits for the key and use the bottom 8 bits to index the uint256 bitmap